pool.go

```diff
@@ -28,26 +31,76 @@ const (
 	tierMainToUnknown  = "main-to-unknown"
 	tierUnknownToMain  = "unknown-to-main"
 	BackendOverrideKey = "CABOOSE_BACKEND_OVERRIDE"
+	CabooseJwtIssuer   = "caboose-client"
 )
 
+// authenticateReq adds authentication to a request when a JWT_SECRET is present as an environment variable.
+func authenticateReq(req *http.Request) (*http.Request, error) {
+
+	// Check for existense of an auth secret.
+	jwtKey := []byte(os.Getenv("JWT_SECRET"))
```

Comment:
we should probably pull this from caboose.Config, rather than only pulling it in as an env var

Reply:
Makes sense, can change that.